cmd/notify-mailer/main.go

```diff
@@ -130,68 +137,68 @@ func (m *mailer) run() error {
 
 	startTime := m.clk.Now()
 
-	for i, dest := range destinations {
-		m.printStatus(dest, i+1, len(destinations), startTime)
-		if strings.TrimSpace(dest) == "" {
+	sortedAddresses := sortAddresses(addressesToRecipients)
+
+	var sent int
+	for i, address := range sortedAddresses {
+		if !m.targetRange.includes(address) {
+			m.log.Debugf("skipping %q: out of target range")
 			continue
 		}
-		err := m.mailer.SendMail([]string{dest}, m.subject, m.emailTemplate)
+		recipients := addressesToRecipients[address]
+		i += 1
+		m.printStatus(address, i, len(addressesToRecipients), startTime)
+		var mailBody bytes.Buffer
+		err = m.emailTemplate.Execute(&mailBody, recipients)
 		if err != nil {
 			switch err.(type) {
 			case bmail.InvalidRcptError:
```

Comment:
I think you have a merge error here that undoes the work from https://github.com/letsencrypt/boulder/commit/1a68cc2225f55ea9612e8563d51f7426576a1578 and replaces it with some dead code.

The `err` that is being type switched in your PR is from `m.emailTemplate.Execute` and could **never** be a `bmail.InvalidRcptError`. I think this whole `switch` should be after L165: `err := m.mailer.SendMail(...)`